hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -154,19 +154,21 @@ private void initKeyGenIfNeeded(boolean populateMetaFields) {
       LOG.info("RDD PreppedRecords was persisted at: " + inputRecordsRDD.getStorageLevel());
     }
 
-    WorkloadProfile profile = null;
+    WorkloadProfile inputProfile = null;
     if (isWorkloadProfileNeeded()) {
       context.setJobStatus(this.getClass().getSimpleName(), "Building workload profile");
-      profile = new WorkloadProfile(buildProfile(inputRecordsRDD), operationType);
-      LOG.info("Workload profile :" + profile);
-      saveWorkloadProfileMetadataToInflight(profile, instantTime);
+      inputProfile = new WorkloadProfile(buildProfile(inputRecordsRDD), operationType);
+      LOG.info("Input workload profile :" + inputProfile);
     }
 
     // handle records update with clustering
     JavaRDD<HoodieRecord<T>> inputRecordsRDDWithClusteringUpdate = clusteringHandleUpdate(inputRecordsRDD);
 
     // partition using the insert partitioner
-    final Partitioner partitioner = getPartitioner(profile);
+    final Partitioner partitioner = getPartitioner(inputProfile);
+    WorkloadProfile executionProfile = ((org.apache.hudi.table.action.commit.Partitioner)partitioner).getExecutionWorkloadProfile();
```

Comment:
I understand isWorkloadProfileNeeded() is always true for all engines, but just to retain old behavior, can we place these two lines within the if block.
```
if (isWorkloadProfileNeeded()) {
  WorkloadProfile executionProfile = ((org.apache.hudi.table.action.commit.Partitioner)partitioner).getExecutionWorkloadProfile();
    saveWorkloadProfileMetadataToInflight(executionProfile, instantTime);
}
```